### PR TITLE
Fix ArrayIndexOutOfBoundsException

### PR DESCRIPTION
**Pull request by dtelaroli**
Details: https://groups.google.com/forum/#!topic/caelum-vraptor/EZliqgan-Bg

### CODE REVIEW

**Comment by Turini** [comment]
@rafaeldff what do you think?

**Comment by garcia-jj** [comment]
@Turini I think the better place to fix this issue is adding a new condition to `NewObject.JavaSetter.settersOf`.

https://github.com/rafaeldff/Iogi/blob/master/src/br/com/caelum/iogi/reflection/NewObject.java#L120

` && method.getParametersLenth() == 1 && method.getName().startsWith("set")`

**Comment by dtelaroli** [comment]
@garcia-jj Fixed, I just followed the error when I was debugging it, but is realy better filter setters.

**Comment by dtelaroli** [comment]
@Turini @rafaeldff @garcia-jj could we release it to use in the next vraptor release?

**Comment by garcia-jj** [comment]
@dtelaroli Iogi is not a Caelum project, that its a personal project by @rafaeldff. So we need to wait @rafaeldff time. 

As my last experiences, sorry for the truth, I feel the project was abandoned by the owner. If you look for last pull request and issues, none are replied by the owner. In the last week I closed #24 because almost 1 year without any response.

https://github.com/rafaeldff/Iogi/pull/19#issuecomment-44415106

**Comment by rafaeldff** [comment]
I don't disagree with @garcia-jj. Hold on for further announcements.

**Comment by Turini** [comment]
@dtelaroli have you tested this change on your project? Everything working as expected?
i will take a closer look on this code on weekend... and if it's ok, a new release on monday.

**Comment by rafaeldff** [comment]
Quick note on project governance: in addition to me, Cecília Fernandes, Lucas Cavalcanti, and now, Rodrito Turini, have commit priviledges, and are welcome to use them! Thank you all for collaborating.

**Comment by dtelaroli** [comment]
@Turini It's working perfect. I tested it.

File: test/br/com/caelum/iogi/vraptor/OgnlParametersProviderTestVRaptorTests.java [hunk 1]
```diff
@@ -125,4 +125,28 @@ public void removeFromTheCollectionIfAnElementIsCreatedWithinACollectionButNoFie
...
+			this.name = name;
+		}
+		
+		public void setEmptyProperty() {
+			
+		}
+	}
+	
+	@Test
+	public void setterWithoutProperty() {
```
Comment (Turini) [inline test/br/com/caelum/iogi/vraptor/OgnlParametersProviderTestVRaptorTests.java]: i'd just rename this test, for `shouldIgnoreMethodsWithSetPrefix` or something like that
what do you think @dtelaroli? besides that, everything looks great. Thank you for your pr

**Comment by Turini** [comment]
> @Turini It's working perfect. I tested it.

Amazing! I left a simple suggestion, after that we can merge it ;)

**Comment by dtelaroli** [comment]
@Turini done

**Comment by Turini** [comment]
amazing! thanks again @dtelaroli! I'll let u know when it's released

**Comment by Turini** [comment]
@dtelaroli I've released iogi 1.0.1 a few hours ago
this is already available on maven: http://repo1.maven.org/maven2/br/com/caelum/iogi/1.0.1/. 
Could you test this version on your project? It'd help a lot :)

**Comment by dtelaroli** [comment]
@Turini It's works fine.
Will you change the vraptor 4.2 to use the new 1.0.1 before release final release?
I'm using the 4.2.0.CR3.

```
<dependency>
	<groupId>br.com.caelum</groupId>
	<artifactId>vraptor</artifactId>
	<version>4.2.0-RC3</version>
	<exclusions>
		<groupId>br.com.caelum</groupId>
		<artifactId>iogi</artifactId>	
	</exclusions>
</dependency>
<dependency>
	<groupId>br.com.caelum</groupId>
	<artifactId>iogi</artifactId>
	<version>1.0.1</version>
</dependency>
```

**Comment by Turini** [comment]
Amazing, thank you @dtelaroli!  Yes, i'll update vraptor using this version from now

**Comment by dtelaroli** [comment]
Excellent. Thank you.

**Pull request merged**